### PR TITLE
add test that uses jdk11

### PR DESCRIPTION
**Pull request by nlopezgi**
* Add detect_java_home as attr to rbe_autoconfig to have users explicitly state if they want java_home to be detected
* add default_java_home attr to toolchain_config_suite_spec to optionally have a fallback if user does not provide one and configs do not exist
* do not validate java for selection of checked-in configs. Since java toolchain can be produced w/o container (if no default_java_home is selected) then no need to use checked-in configs for this. 
* add unit tests for use of detect_java_home
* add basic GCB test that uses jdk11

Pending for upcoming PR: use jdk11 in RBE test that builds Bazel

### CODE REVIEW

File: rules/rbe_repo.bzl [hunk 1]
```diff
@@ -845,6 +856,9 @@ def rbe_autoconfig(
       digest: Optional. The digest of the image to pull.
           Should not be set if 'tag' is used.
           Must be set together with 'registry' and 'repository'.
+      detect_java_home: Optional. Default False. Should only be set
```
Comment (xingao267) [inline rules/rbe_repo.bzl]: If java_home is not specified, and detect_java_home is false, then the java_home in java_runtime() will be empty? or the java config will not be create at all?

File: rules/rbe_repo.bzl [hunk 2]
```diff
@@ -898,8 +910,11 @@ def rbe_autoconfig(
     if bazel_rc_version and not bazel_version:
         fail("bazel_rc_version can only be used with bazel_version.")
 
-    if not create_java_configs and java_home != None:
-        fail("java_home should not be set when create_java_configs is false.")
+    if not create_java_configs and (java_home != None or detect_java_home):
```
Comment (xingao267) [inline rules/rbe_repo.bzl]: if not create_java_configs and (java_home or detect_java_home):

File: rules/rbe_repo.bzl [hunk 3]
```diff
@@ -898,8 +910,11 @@ def rbe_autoconfig(
...
+        fail("java_home / detect_java_home should not be set when " +
+             "create_java_configs is False.")
+    if java_home != None and detect_java_home:
```
Comment (xingao267) [inline rules/rbe_repo.bzl]: if java_home and detect_java_home

File: rules/rbe_repo.bzl (hunk 1 above)
Comment (xingao267) [reply rules/rbe_repo.bzl]: Saw the fallback logic below. But that's kinda complicated...

File: rules/rbe_repo.bzl [hunk 4]
```diff
@@ -943,6 +958,28 @@ def rbe_autoconfig(
         use_checked_in_confs = use_checked_in_confs,
     )
 
+    # If create_java_configs was requested but no java_home or detect_java_home was
+    # set, we try to resolve a java_home
+    if create_java_configs and not java_home and not detect_java_home:
+        # If a spec was found and that has a java_home, use it
+        if toolchain_config_spec and toolchain_config_spec.create_cc_configs:
```
Comment (xingao267) [inline rules/rbe_repo.bzl]: toolchain_config_spec.create_java_configs?

File: rules/rbe_repo.bzl [hunk 5]
```diff
@@ -943,6 +958,28 @@ def rbe_autoconfig(
...
+            # Fallback to try to using the default_java_home set in the
+            # toolchain_config_suite_spec
+            java_home = toolchain_config_suite_spec.get("default_java_home")
+
+        elif use_checked_in_confs != CHECKED_IN_CONFS_FORCE:
+            # Fallback to detecting the java_home if CHECKED_IN_CONFS_FORCE
+            # was not passed
+            detect_java_home = True
+
+        elif toolchain_config_spec and use_checked_in_confs == CHECKED_IN_CONFS_FORCE:
```
Comment (xingao267) [inline rules/rbe_repo.bzl]: Not fully understand this part.

File: rules/rbe_repo.bzl (hunk 2 above)
Comment (nlopezgi) [reply rules/rbe_repo.bzl]: fixed

File: rules/rbe_repo.bzl (hunk 3 above)
Comment (nlopezgi) [reply rules/rbe_repo.bzl]: fixed

File: rules/rbe_repo.bzl (hunk 4 above)
Comment (nlopezgi) [reply rules/rbe_repo.bzl]: fixed

File: rules/rbe_repo.bzl (hunk 5 above)
Comment (nlopezgi) [reply rules/rbe_repo.bzl]: changed to a fail early and documented better, thanks for noting

File: rules/rbe_repo.bzl (hunk 1 above)
Comment (nlopezgi) [reply rules/rbe_repo.bzl]: added notes about fallback logic. ptal

**Pull request merged**